lib/internal/url.js

```diff
@@ -254,8 +254,262 @@ class URLSearchParams {
+    const len = a.length;
+
+    if (len <= 2) {
+      // Nothing needs to be done.
+    } else if (len < 100) {
+      // 100 is found through testing.
+      // Simple stable in-place insertion sort
+      // Derived from v8/src/js/array.js
+      for (let i = 2; i < len; i += 2) {
+        const curKey = a[i];
+        const curVal = a[i + 1];
+        let j;
+        for (j = i - 2; j >= 0; j -= 2) {
+          if (a[j] > curKey) {
+            a[j + 2] = a[j];
+            a[j + 3] = a[j + 1];
+          } else {
+            break;
+          }
+        }
+        a[j + 2] = curKey;
+        a[j + 3] = curVal;
+      }
+    } else {
+      // Bottom-up iterative stable merge sort
+      const lBuffer = new Array(len);
+      const rBuffer = new Array(len);
+      for (let step = 2; step < len; step *= 2) {
+        for (let start = 0; start < len - 2; start += 2 * step) {
+          const mid = start + step;
+          let end = mid + step;
+          end = end < len ? end : len;
+          if (mid > end)
+            continue;
+          merge(a, start, mid, end, lBuffer, rBuffer);
+        }
+      }
+    }
+
+    update(this[context], this);
+  }
+
+  // https://heycam.github.io/webidl/#es-iterators
+  // Define entries here rather than [Symbol.iterator] as the function name
+  // must be set to `entries`.
+  entries() {
+    if (!this || !this[searchParams] || this[searchParams][searchParams]) {
+      throw new ERR_INVALID_THIS('URLSearchParams');
+    }
+
+    return createSearchParamsIterator(this, 'key+value');
+  }
+
+  forEach(callback, thisArg = undefined) {
+    if (!this || !this[searchParams] || this[searchParams][searchParams]) {
+      throw new ERR_INVALID_THIS('URLSearchParams');
+    }
+    validateCallback(callback);
+
+    let list = this[searchParams];
+
+    let i = 0;
+    while (i < list.length) {
+      const key = list[i];
+      const value = list[i + 1];
+      callback.call(thisArg, value, key, this);
+      // In case the URL object's `search` is updated
+      list = this[searchParams];
+      i += 2;
+    }
+  }
+
+  // https://heycam.github.io/webidl/#es-iterable
+  keys() {
+    if (!this || !this[searchParams] || this[searchParams][searchParams]) {
+      throw new ERR_INVALID_THIS('URLSearchParams');
+    }
+
+    return createSearchParamsIterator(this, 'key');
+  }
+
+  values() {
+    if (!this || !this[searchParams] || this[searchParams][searchParams]) {
+      throw new ERR_INVALID_THIS('URLSearchParams');
+    }
+
+    return createSearchParamsIterator(this, 'value');
+  }
+
+  // https://heycam.github.io/webidl/#es-stringifier
+  // https://url.spec.whatwg.org/#urlsearchparams-stringification-behavior
+  toString() {
+    if (!this || !this[searchParams] || this[searchParams][searchParams]) {
+      throw new ERR_INVALID_THIS('URLSearchParams');
+    }
+
+    return serializeParams(this[searchParams]);
+  }
 }
 
+ObjectDefineProperties(URLSearchParams, {
+  append: { enumerable: true },
```

Comment:
I think all of them should be also `configureable: true`. Or is there a reason why they are only enumerable?

Reply:
`configurable: true` is only necessary when adding new properties.